File: typescript/day-12/part-2.ts
import { loadInput, sum, toInt } from "../utils";


// Todo: put test_input into a "test" file.
const test_input = `???.### 1,1,3
.??..??...?##. 1,1,3
?#?#?#?#?#?#?#? 1,3,1,6
????.#...#... 4,1,1
????.######..#####. 1,6,5
?###???????? 3,2,1`;


// const input = test_input;
const input = loadInput("input");
const rows = input.split("\n").filter(Boolean);

console.log(rows);

File: kotlin/src/main/kotlin/day01/part-2.kt
package day01.part2

import loadInput

fun main() {
    val input = loadInput("./src/main/kotlin/day01/input")

    val words = listOf("one", "two", "three", "four", "five", "six", "seven", "eight", "nine")

    // wip p1
    val solution = input.lines().map { it ->
        it.split("")
            .filter {
                it.toIntOrNull() !== null
            }.map {
                it.toInt()
            }
    }.sumOf { nums ->
        "${nums.first()}${nums.last()}".toInt()
    }

    println(solution)
}
